dev/index.tsx

```diff
@@ -1,21 +1,120 @@
-import React from "react";
+import React, { ChangeEvent, useCallback, useState } from "react";
 import ReactDOM from "react-dom";
 import { ReactSearch } from "../";
 
-const App = () => (
-  <div id="app">
-    <h1>@vectara/react-search</h1>
-    <div id="searchWrapper">
-      <ReactSearch
-        corpusId="1"
-        customerId="1366999410"
-        apiKey="zqt_UXrBcnI2UXINZkrv4g1tQPhzj02vfdtqYJIDiA"
-        placeholder="What would you like to search for?"
-      />
-    </div>
-    <code>{codeSnippet}</code>
-  </div>
-);
+const DEFAULT_CORPUS_ID = "1";
+const DEFAULT_CUSTOMER_ID = "1366999410";
+const DEFAULT_API_KEY = "zqt_UXrBcnI2UXINZkrv4g1tQPhzj02vfdtqYJIDiA";
+const DEFAULT_PLACEHOLDER = "What would you like to search for?";
+
+const App = () => {
+  const [corpusId, setCorpusId] = useState<string>("");
+  const [customerId, setCustomerId] = useState<string>("");
+  const [apiKey, setApiKey] = useState<string>("");
+  const [placeholder, setPlaceholder] = useState<string>(DEFAULT_PLACEHOLDER);
+
+  const onUpdateCorpusId = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+    setCorpusId(e.target.value);
+  }, []);
+
+  const onUpdateCustomerId = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+    setCustomerId(e.target.value);
+  }, []);
+
+  const onUpdateApiKey = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+    setApiKey(e.target.value);
+  }, []);
+
+  const onUpdatePlaceholder = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+    setPlaceholder(e.target.value);
+  }, []);
+
+  return (
+    <>
+      <Navbar />
+      <div className="pageContentWrapper">
+        <h1 id="title">@vectara/react-search</h1>
+        <div>
+          <div className="subsection">
+            <p>
+              ReactSearch is the easiest way to add Vectara semantic search to your React apps. With just a few lines of
+              code, you can add a Vectara-powered search UI to your React applications.
+            </p>
+            <p>
+              Try it out below! You can also make changes to the options in the Configuration section to modify the
+              component's behavior/appearance.
+            </p>
+          </div>
+          <div className="searchOuterWrapper">
+            <div className="searchInnerWrapper">
+              <ReactSearch
+                corpusId={corpusId === "" ? DEFAULT_CORPUS_ID : corpusId}
+                customerId={customerId === "" ? DEFAULT_CUSTOMER_ID : customerId}
+                apiKey={apiKey === "" ? DEFAULT_API_KEY : apiKey}
```

Comment:
Here we ensure that if the field is blank, we use the default props that point to the docs page. This ensures that we don't voluntarily display the docs corpus details in the text fields.

Reply:
Might be worth leaving this as a code comment.